examples/rust-outbound-pg/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+use anyhow::{anyhow, Result};
+use spin_sdk::{
+    http::{internal_server_error, Request, Response},
+    http_component, pg,
+};
+
+// The environment variable set in `spin.toml` that points to the
+// address of the Pg server that the component will write to
+const DB_URL_ENV: &str = "DB_URL";
+
+#[http_component]
+fn read(_req: Request) -> Result<Response> {
+    let address = std::env::var(DB_URL_ENV)?;
+
+    let sql = "select * from articletest";
+    let rows = pg::query(&address,sql, &vec![]).map_err(|_| anyhow!("Error execute pg command"))?;
+
+    println!("rows: {:?}", rows);
```

Comment:
Could we give an example here (or start some draft documentation) of how to use payloads?  I don't have an environment to run this but reading the code it looks like the payloads are raw bytes, and it's not obvious to me how to translate those into the component language (Rust, Go, Swift, etc.).

(I'm also a bit unsure about identifying columns in a `select *` query like this, but that would be easy to work around in a production sample by listing the columns explicitly.)

Despite this nit I am very excited to see this!  Thank you for putting it together!

Reply:
Okay, let me look into it later.

Reply:
> Could we give an example here (or start some draft documentation) of how to use payloads? I don't have an environment to run this but reading the code it looks like the payloads are raw bytes, and it's not obvious to me how to translate those into the component language (Rust, Go, Swift, etc.).
> 
> (I'm also a bit unsure about identifying columns in a `select *` query like this, but that would be easy to work around in a production sample by listing the columns explicitly.)
> 
> Despite this nit I am very excited to see this! Thank you for putting it together!

Hi, I updated the example code and add some type converts for the result vec<u8> payload. Please help me to check on it.

Appreciated for your kind suggestions.

Reply:
This is super helpful - thanks!  However I think consumers are still going to need some guidance on how to convert other types e.g. numerics, booleans, SQL NULLs, etc. (I'm not familiar with the full range of Postgres types).  They don't need to be all in the sample (that would be a bit much!) but they need to be in the documentation somewhere.  (If you want, feel free to just jot things down, and we can turn them into public documentation.)

Do you think it's worth baking those conversion helpers into the SDKs?  I guess we can always add them over time!

Reply:
Hi, thanks for your comments.

The full support against to the postgres types is not the task of this PR, in my thought, later I will look into the crate postgres-types, suppose it will cover the abstraction and convertions  between the lower bytes payload and the upper rust types, I wish we could use it in an automatically converting style, just like rust-postgres does.

That's a lot of work, but if you want that, I think I can do it later based on the current work of this PR.

Regards.

Reply:
And yes, I think it's better to implement the convertor layer in the rust SDK rather than in the host component layer. Cause that

- the complexity of the *.wit file;
- the updated frequency of the host component;
- the potential other language SDK must recognize fully the output of the host component.

Reply:
Thanks for the reply.  I'm okay with a "merge and iterate" strategy - happy to get this in and get things motoring.  Thanks!